MazerPlatformer/Character.cs

```diff
@@ -70,20 +73,24 @@ public override void Initialize()
             Animation = new Animation(Animation.AnimationDirection.Down); // AnimationDirection is different to CharacterDirection
 
             // Initialize the characters animation
-            Animation.Initialize(AnimationInfo.Texture, GetCentre(), AnimationInfo.FrameWidth, AnimationInfo.FrameHeight,
-                AnimationInfo.FrameCount, AnimationInfo.Color, AnimationInfo.Scale, AnimationInfo.Looping,
-                AnimationInfo.FrameTime);
+            return EnsureWithReturn( () =>
+            {
+                Animation.Initialize(AnimationInfo.Texture, GetCentre(), AnimationInfo.FrameWidth,
```

Comment:
The problem why this code is inherently unsafe if because it could throw exceptions and we've not know why. Throwing exceptions would cause our function not to honour its return type. (**key**)
- The reason this is unsafe if because we don't now if anything variables will be null here, we've been unable to verify this. These all come in from the locations on the underlying class(member variables) so we've got no control over how they got there and what state they are in and so can't be sure that they will work. 

The solution is to have a say and control the creation or use of these variables si that we can make guaranteed that they will work. A possible solution is to avoid fetching variables from the underlying class, and enforce that they come in as parameters. This way the parameters can at least check that they are not null (force the parameters to be of type Option<> which can't hold parameters). problem with this is that it doesn't ensure that the parameters wont throw any other types of exceptions besiding NullReferenceException....